FAIL: test_retrieve_from_retired_backend

<!-- ⚠️ If you do not respect this template, your issue will be closed -->
<!-- ⚠️ Make sure to browse the opened and closed issues -->

### Information

- **Qiskit IBM Provider version**:
- **Python version**:
- **Operating system**:

### What is the current behavior?
This [Cron-test](https://github.com/Qiskit-Partners/qiskit-ibm/runs/3716585082?check_suite_focus=true#step:5:1354) failed with 
```
======================================================================
FAIL: test_retrieve_from_retired_backend (test.ibm.test_ibm_job.TestIBMJob)
Test retrieving a job from a retired backend.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/qiskit-ibm/qiskit-ibm/test/ibm/test_ibm_job.py", line 448, in test_retrieve_from_retired_backend
    self.assertIn(new_job.job_id(), last_month_job_ids)
AssertionError: '615150b4d9c01f1d8bf8befa' not found in ['615150e3c5770cfe7a2d2125', '615150e3d9c01fd934f8befb', '615150e22e79b34791d362cd', '615150e143669acf282830f8', '615150e03799efc5c6dafc86', '615150ddc5770c50bd2d2124', '615150d22e79b30d73d362cb', '615150cc1281a5fb01b5e3e1', '615150c143669a63f52830f5', '615150c043669a3cfd2830f4']

----------------------------------------------------------------------
```


### Steps to reproduce the problem



### What is the expected behavior?



### Suggested solutions

--- a/test/ibm/test_ibm_job.py
+++ b/test/ibm/test_ibm_job.py
@@ -443,7 +443,8 @@ class TestIBMJob(IBMTestCase):
             new_job = self.provider.backend.job(self.sim_job.job_id())
             self.assertTrue(isinstance(new_job.backend(), IBMRetiredBackend))
             self.assertNotEqual(new_job.backend().name(), 'unknown')
-            last_month_jobs = self.provider.backend.jobs(start_datetime=self.last_month)
+            last_month_jobs = self.provider.backend.jobs(
+                start_datetime=self.last_month, backend_name=new_job.backend().name())
             last_month_job_ids = [job.job_id() for job in last_month_jobs]
             self.assertIn(new_job.job_id(), last_month_job_ids)
         finally:

